src/docs/sphinx/dev_guide/updating_tpls.rst

```diff
@@ -0,0 +1,221 @@
+
+They can be found in our repository here: `src/thirdparty`. The basic 
+instructions on how to update a built-in TPL are as follows:
+
+#. Download the new release and override the source that is already there.
+   This can often involve removing files no-longer needed but most of the
+   current ones are a single header file.
+
+#. Review and apply the existing patch files. More than likely, you will not
+   be able to directly apply the patch but it will give you the general idea
+   on what needs to be applied.  For example, the namespace update mentioned above.
+
+#. Ensure that the build and tests still pass. More info on :ref:`testing-label`.
+
+#. Follow the normal pull request work flow. More info on :ref:`pullrequest-label`.
+
+
+.. _local-tpls-label:
+
+Local Third-party Library Installation
+--------------------------------------
+
+It is often useful to have a different set of TPLs during the development process.
+For example, you may want to try out a new library or version of an existing library.
+
+From the top-level Axom directory, run the following script to build all TPLs
+for all existing compiler specs on the platform you are currently on::
+
+$ ./scripts/llnl_scripts/build_tpls.py -d local/install/path
+
+where ``local/install/path`` is a directory location where you want the 
+libraries to be installed.
+
+It will output whether the TPL install succeeded and, 
+subsequently, whether an Axom build against the TPL install succeeded.
+
+Running the script produces new host-config files (i.e., CMake cache files) 
+that you can use to build and test Axom with the installation, if issues
+arise. The generated host-config files will be located in the top-level Axom
+directory of your local clone of the repo. If any changes to Axom code are 
+needed to work with the TPL update(s), make the changes and test them.
+
+.. note:: You can build a subset of TPLs for a platform, by passing a Spack
+          spec arguments to the `build_tpls.py` script. For example,
+
+          ``--spec clang@10.0.0~cpp14+devtools+mfem+c2c``
+
+          will build the TPLs for the clang 10.0.0 compiler. Please see the
+          ``scripts/spack/specs.json`` file for a list of currently tested specs. 
+
+
+Shared Third-party Library Installation Steps
+---------------------------------------------
+
+The following instructions describe how to install local copies of Axom
+TPLs on Livermore Computing (LC) platforms and recreate our Docker containers
+with a new set of TPLs. Typically, this process is followed when you want to 
+update one or more TPLs which Axom depends on. After they are built and
+the required changes are merged into develop, they will be available for
+other Axom developers to use during development, in Axom Gitlab CI testing, etc.
+
+#. **Working on a local branch.** 
+   Make a local clone of the Axom repo and create a branch to work on. Working
+   on a separate branch is a good practice to avoid propagating mis-steps
```

Comment:
Is the second sentence needed? Our main and develop branches are protected on GitHub to prevent accidental pollution.